application/src/main/java/org/opentripplanner/transit/service/TransitService.java

```diff
@@ -269,6 +270,17 @@ List<TripTimeOnDate> findTripTimeOnDate(
     boolean includeCancellations
   );
 
+  /**
+   * Fetch upcoming vehicle departures from a stop for a specific pattern, passing the stop for the
+   * previous, current and next service date. It uses a priority queue to keep track of the next
+   * departures. The queue is shared between all dates, as services from the previous service date
+   * can visit the stop later than the current service date's services.
+   * <p>
+   * This method is similar to {@link TransitService#findTripTimesOnDate(StopLocation, TripPattern, Instant, Duration, int, ArrivalDeparture, boolean)}
+   * in that it uses a filter request which allows you to include and exclude routes, agencies and modes.
+   */
+  List<TripTimeOnDate> findTripTimesOnDate(TripTimeOnDateRequest request);
```

Comment:
Not sure if it works, but could return a `TripTimeOnDateList`, then the list could have the section logic.

Reply:
In light of what we talked about in the dev meeting (this being as DTO and not a domain model) do you still think I should move these methods to a factory?